packages/react-i18n/src/utilities/translate.tsx

```diff
@@ -193,104 +202,106 @@ function translateWithDictionary(
   }
 
   const processedString =
-    typeof result === 'string' && pseudotranslate
+    isString(result) && pseudotranslate
       ? pseudotranslateString(result, {
           ...PSEUDOTRANSLATE_OPTIONS,
           toLocale:
             typeof pseudotranslate === 'boolean' ? undefined : pseudotranslate,
         })
       : result;
 
-  if (typeof processedString === 'string') {
-    return updateStringWithReplacements(processedString, {
-      ...replacements,
-      ...additionalReplacements,
-    });
-  } else {
+  if (!isString(processedString)) {
     return MISSING_TRANSLATION;
   }
+
+  return updateStringWithReplacements(
+    processedString,
+    {
+      ...replacements,
+      ...additionalReplacements,
+    },
+    {interpolate},
+  );
 }
 
+type UpdateStringWithReplacementsOptions = Pick<
+  TranslateOptions,
+  'interpolate'
+>;
+
 function updateStringWithReplacements(
   str: string,
   replacements?: ComplexReplacementDictionary,
+  options?: UpdateStringWithReplacementsOptions,
 ): React.ReactElement<any>;
 function updateStringWithReplacements(
   str: string,
   replacements?: PrimitiveReplacementDictionary,
+  options?: UpdateStringWithReplacementsOptions,
 ): string;
 function updateStringWithReplacements(
   str: string,
   replacements:
     | ComplexReplacementDictionary
     | PrimitiveReplacementDictionary = {},
+  {interpolate}: UpdateStringWithReplacementsOptions = {},
 ): any {
-  const replaceFinder = /([^{]*)({([^}]*)})?/g;
-  const allReplacementsArePrimitives = Object.keys(replacements).every(
-    (key) => typeof replacements[key] !== 'object',
-  );
-
-  if (allReplacementsArePrimitives) {
-    return str.replace(REPLACE_REGEX, (match) => {
-      const replacement = match.substring(1, match.length - 1);
-
-      if (!Object.prototype.hasOwnProperty.call(replacements, replacement)) {
-        throw new MissingReplacementError(replacement, replacements);
+  const pieces: (string | React.ReactElement<any>)[] = [];
+
+  const replaceFinder = new RegExp(interpolate || DEFAULT_INTERPOLATION, 'g');
+
+  let matchIndex = 0;
+  let lastOffset = 0;
+
+  // Uses replace callback, but not its return value
+  str.replace(
+    replaceFinder,
+    (
+      match,
+      _fullPlaceholder: string,
```

Comment:
Instead of `_fullPlaceholder`, could we remove the outer capture group for each regex? Does that make sense / why does `_fullPlaceholder` exist?